core/controllers/contributor_dashboard.py

```diff
@@ -361,54 +360,34 @@ def _get_reviewable_exploration_opportunity_summaries(
         Returns:
             list(ExplorationOpportunitySummary). A list of the matching
             exploration opportunity summaries.
-
-        Raises:
-            Exception. No exploration_id found for the node_id.
         """
-        # 1. Fetch the eligible topics.
-        # 2. Fetch the stories for the topics.
-        # 3. Get the reviewable translation suggestion target IDs for the user.
-        # 4. Get story exploration nodes in order, filtering for explorations
-        # that have in review translation suggestions.
-        if topic_name is None:
-            topics = topic_fetchers.get_all_topics()
+        # 1. Fetch all the exploration IDs of published chapters in published
+        #    stories under the topic(s).
+        # 2. Fetch each reviewable translation suggestion to be available to the
+        #    translation reviewer.
+        # 3. Filter exploration ids. Keep those that are the target of any
+        #    fetched reviewable translation suggestions.
+        # 4. Fetch all exploration opportunity summaries for the kept
+        #    explorations.
+        # 5. If there's a pinned lesson, move the pinned lesson's summary to
+        #    the top.
+        if topic_name is None or topic_name == 'undefined':
```

Comment:
Why would 'undefined' be passed? We should only accept None and regard 'undefined' as a topic whose name is 'undefined'. If the frontend passes undefined then that is incorrect and we should fix it.

Reply:
Okay, so I changed it so that it now fetches all topics when topic_name is either None or an empty string (since no topic can have that empty string as a name).

Reply:
Ah, don't do that -- we should treat "empty string" as a topic whose name is the empty string, otherwise this relies on hidden conventions. None (or null) should be the only allowed value for the "all" case.

Is there a reason why we need to handle the empty string case (in what circumstances would it get passed in)?

Reply:
A topic name cannot be an empty string, so there's no reason to handle the empty string case.

Also, does the backend convert JSON's `null` to Python's `None`?

Reply:
> Also, does the backend convert JSON's null to Python's None?

Yup, it does.

Reply:
Done. Now the frontend sends `null` when `topicName` is `null` or `undefined`. The backend receives `null` (as a string, which means the backend isn't converting `null` to `None` automatically), and translates `'null'` to `None`. Although because of this, a topic whose name is `'null'` will result in the app fetching all topics instead.

Reply:
On the backend, the `BaseHandler` class initializes its `normalized_request` dict values by `request.get('topic_name')` (from webapp2). I logged the value resulting from `request.get('topic_name')`, which comes out as the string `'null'`. So that tells me that the topic name is `'null'` even when executing code found in the `BaseHandler` and webapp2.

I also found the only `json.loads()` call in https://github.com/oppia/oppia/blob/develop/core/controllers/base.py#L202, which supposedly translates JSON `null` to Python `None`. In this context, `self.request.get('payload')` several lines above seems to return nothing when I try to log its value.

Reply:
Ah, sorry, didn't realize this was a GET handler. You'd probably need to use '' and do something like `request.get(..., None)` (should double-check if that works, not sure if empty string results in the default value being used; if not an `if '' then set to None` should be OK). 

Suggest leaving a comment to explain what's going on here, too (i.e. that this is a GET request so all stuff comes in as strings.)

Reply:
Done. I went with `request.get(..., None)`.